modules/data.land/tests/testthat/test-extract_soil_nc.R

```diff
@@ -0,0 +1,343 @@
+    depths = c(0.15, 0.30, 0.60, 1.0)
+  )
+  if (!is.null(res_single)) {
+    expect_true(all(file.exists(unlist(res_single))))
+  }
+  if (!is.null(res_multiple)) {
+    expect_true(all(file.exists(unlist(res_multiple))))
+  }
+  unlink(tmp_outdir, recursive = TRUE)
+})
+
+test_that("extract_soil_gssurgo handles edge cases in soil data processing", {
+  skip_on_cran()
+  skip_on_ci()
+  lat <- 40.0
+  lon <- -88.0
+  tmp_outdir <- tempfile("gssurgo_test_")
+  dir.create(tmp_outdir, recursive = TRUE)
+  
+  # Test with minimal grid (single point)
+  res <- extract_soil_gssurgo(
+    outdir = tmp_outdir,
+    lat = lat,
+    lon = lon,
+    size = 1,
+    grid_size = 1,
+    grid_spacing = 1,
+    depths = c(0.15)
+  )
+  
+  if (!is.null(res)) {
+    expect_true(is.list(res) || is.vector(res))
+    expect_true(all(file.exists(unlist(res))))
+  }
+  unlink(tmp_outdir, recursive = TRUE)
+})
+
+test_that("extract_soil_gssurgo maintains data consistency across ensemble members", {
+  skip_on_cran()
+  skip_on_ci()
+  if (!requireNamespace("ncdf4", quietly = TRUE)) {
+    skip("ncdf4 package not available")
+  }
+  lat <- 40.0
+  lon <- -88.0
+  tmp_outdir <- tempfile("gssurgo_test_")
+  dir.create(tmp_outdir, recursive = TRUE)
+  
+  res <- extract_soil_gssurgo(
+    outdir = tmp_outdir,
+    lat = lat,
+    lon = lon,
+    size = 3,
+    grid_size = 3,
+    grid_spacing = 10,
+    depths = c(0.15, 0.30)
+  )
+  
+  if (!is.null(res) && length(res) > 1) {
+    file_paths <- unlist(res)
+    # Check that all files have same structure
+    first_nc <- ncdf4::nc_open(file_paths[1])
+    first_vars <- names(first_nc$var)
+    first_dims <- sapply(first_nc$var, function(v) v$size)
+    ncdf4::nc_close(first_nc)
+    for (i in 2:length(file_paths)) {
+      nc <- ncdf4::nc_open(file_paths[i])
+      vars <- names(nc$var)
+      dims <- sapply(nc$var, function(v) v$size)
+      ncdf4::nc_close(nc)
+      expect_equal(vars, first_vars, 
+                  info = paste("Variable names differ between ensemble members"))
+      expect_equal(dims, first_dims, 
+                  info = paste("Dimension sizes differ between ensemble members"))
+    }
+  }
+  unlink(tmp_outdir, recursive = TRUE)
+})
+
+test_that("extract_soil_gssurgo returns NULL when no valid soil data is found", {
+  skip_on_cran()
+  skip_on_ci()
+  lat <- 0.0
+  lon <- 0.0
+  tmp_outdir <- tempfile("gssurgo_test_")
+  dir.create(tmp_outdir, recursive = TRUE)
+  
+  res <- extract_soil_gssurgo(
+    outdir = tmp_outdir,
+    lat = lat,
+    lon = lon,
+    size = 1,
+    grid_size = 3,
+    grid_spacing = 10,
+    depths = c(0.15)
+  )
+  expect_null(res) # Error: No mapunit keys were found for this site.
+  unlink(tmp_outdir, recursive = TRUE)
+})
+
+test_that("extract_soil_gssurgo performance is reasonable for typical use cases", {
+  skip_on_cran()
+  skip_on_ci()
+  lat <- 40.0
+  lon <- -88.0
+  tmp_outdir <- tempfile("gssurgo_test_")
+  dir.create(tmp_outdir, recursive = TRUE)
```

Comment:
We can let withr handle creating this and cleaning it up for us when the test finishes:

```suggestion
  tmp_outdir <- withr::local_tempdir("gssurgo_test_")
```

Reply:
(Probably obvious, but: I'm only commenting here but you should use this same pattern in all the test blocks that currently have a `dir.create`)

Reply:
fixed 👍